stdlib/public/core/RangeReplaceableCollection.swift

```diff
@@ -453,11 +453,18 @@ extension RangeReplaceableCollection {
   @inlinable
   public mutating func append<S: Sequence>(contentsOf newElements: __owned S)
     where S.Element == Element {
-
-    let approximateCapacity = self.count + newElements.underestimatedCount
-    self.reserveCapacity(approximateCapacity)
-    for element in newElements {
-      append(element)
+    
+    let done:Void? = newElements.withContiguousStorageIfAvailable {
+      _onFastPath()
+      replaceSubrange(endIndex..<endIndex, $0)
+    }
+      
+    if done == nil {
+      let approximateCapacity = self.count + newElements.underestimatedCount
+      self.reserveCapacity(approximateCapacity)
```

Comment:
Note for followup work: This `reserveCapacity` call is extremely undesirable, for the same reason `append(_:)` doesn't call `reserveCapacity(self.count + 1)`: for contiguous collections, `reserveCapacity` does not use the usual exponential resizing logic, it allocates precisely as much memory as needed. This makes calling `append(contentsOf:)` in a loop algorithmically worse than if `reserveCapacity` wasn't used: in the worst case, it can cause using `append(contentsOf:)` n times to append constant-size blocks to reallocate storage _on every single invocation_, rather than log(n) times.